### PR TITLE
Expose success url and realm in getSessionToken

### PR DESCRIPTION
**Pull request by vahancouver**
[SDKS-3352](https://bugster.forgerock.org/jira/browse/SDKS-3352)

[iOS] - Expose realm, Success  URL in GetSessionToken

### CODE REVIEW

**Review by rodrigoareis — APPROVED** [review]
Changes looks good to me

File: FRAuth/FRAuth/Authentication/AuthService.swift [hunk 1]
```diff
@@ -229,7 +229,9 @@ public class AuthService: NSObject {
                     }
                 }
                 else if let tokenId = response[OpenAM.tokenId] as? String {
```
Comment (jeyanthanperiyasamy) [inline FRAuth/FRAuth/Authentication/AuthService.swift:231]: Does the user has a was way to get the failureURL on any Exception.

File: FRAuth/FRAuthTests/FRAuthSwiftTests/FRAuth/Model/OAuth2/TokenTests.swift [hunk 2]
```diff
@@ -12,14 +12,27 @@ import XCTest
 @testable import FRAuth
 
 class TokenTests: FRAuthBaseTest {
-    
+    //= SSOToken("mySSOToken", "http://success.url", "myRealm")
```
Comment (jeyanthanperiyasamy) [inline FRAuth/FRAuthTests/FRAuthSwiftTests/FRAuth/Model/OAuth2/TokenTests.swift:15]: do we need this ?

**Review by jeyanthanperiyasamy — APPROVED** [review]
some minor comments ..

File: FRAuth/FRAuthTests/FRAuthSwiftTests/FRAuth/Model/OAuth2/TokenTests.swift (hunk 2 above)
Comment (vahancouver) [reply FRAuth/FRAuthTests/FRAuthSwiftTests/FRAuth/Model/OAuth2/TokenTests.swift]: Removed

File: FRAuth/FRAuth/Authentication/AuthService.swift (hunk 1 above)
Comment (vahancouver) [reply FRAuth/FRAuth/Authentication/AuthService.swift:231]: @witrisna Do you have an answer for this?

Comment (witrisna) [reply FRAuth/FRAuth/Authentication/AuthService.swift:231]: When failed with the Journey, Android throws AuthenticationException with the raw body result, the raw body should contains the failureURL.

**Review by witrisna — APPROVED** [review]

File: FRAuth/FRAuth/Manager/KeychainManager.swift [hunk 3]
```diff
@@ -30,6 +30,8 @@ struct KeychainManager {
         case ssoToken = "sso_token"
         case primaryService = "primaryService"
         case primaryServiceEncrypted = "primaryService-encrypted"
+        case successUrl = "successUrl"
+        case realm = "relam"
```
Comment (george-bafaloukas-forgerock) [inline FRAuth/FRAuth/Manager/KeychainManager.swift:34]: Typo on `relam`

File: FRAuth/FRAuthTests/FRAuthSwiftTests/FRAuth/Model/OAuth2/TokenTests.swift [hunk 4]
```diff
@@ -63,6 +75,39 @@ class TokenTests: FRAuthBaseTest {
         }
     }
     
+    func testTokenSecureCodingWithoutDefaultValues() {
+        if #available(iOS 11.0, *) {
```
Comment (george-bafaloukas-forgerock) [inline FRAuth/FRAuthTests/FRAuthSwiftTests/FRAuth/Model/OAuth2/TokenTests.swift:79]: Why 11? We are on 12 going to 13 now right? We might be able to up this a bit if not mistaken

**Review by george-bafaloukas-forgerock — CHANGES_REQUESTED** [review]
Change the typo please. Other than that all good

**Review by spetrov — APPROVED** [review]
Looks good to me!

**Review by spetrov — APPROVED** [review]

File: FRAuth/FRAuthTests/FRAuthSwiftTests/FRAuth/Model/OAuth2/TokenTests.swift (hunk 4 above)
Comment (spetrov) [reply FRAuth/FRAuthTests/FRAuthSwiftTests/FRAuth/Model/OAuth2/TokenTests.swift]: This has been addressed...

File: FRAuth/FRAuth/Manager/KeychainManager.swift (hunk 3 above)
Comment (spetrov) [reply FRAuth/FRAuth/Manager/KeychainManager.swift]: This has been addressed...

**Pull request merged**